File: src/scripts/check-recommendations.ts
import dotenv from "dotenv";
import { MongoClient } from "mongodb";

dotenv.config();

async function main() {
  const tableId = process.argv[2] ?? "T-0019";
  const uri = process.env.MONGODB_URI;
  if (!uri) throw new Error("MONGODB_URI not set");
  const client = new MongoClient(uri);
  await client.connect();
  const db = client.db(process.env.MONGODB_DB ?? "casino_marketing_demo");

  const recs = await db
    .collection("table_minbet_recommendations")
    .find({ tableId }, { projection: { _id: 0, recommendationId: 1, status: 1, createdAt: 1, drivers: 1, currentMinBet: 1, recommendedMinBet: 1 } })
    .sort({ createdAt: -1 })
    .limit(5)
    .toArray();

  console.log(`Recent recommendations for ${tableId}:`);
  if (recs.length === 0) {
    console.log("  (none)");
  }
  for (const r of recs) {
    console.log(
      `  ${r.recommendationId}  status=${r.status}  current=${r.currentMinBet}→${r.recommendedMinBet}  occupancy=${r.drivers?.occupancyRate}  createdAt=${new Date(r.createdAt).toLocaleString()}`
    );
  }
  await client.close();
}
main();

File: server/app/scripts/check_recommendations.py
"""Print the 5 most recent min-bet recommendations for a given table.

Run with::

    uv run python -m app.scripts.check_recommendations              # T-0019
    uv run python -m app.scripts.check_recommendations T-0007       # custom
"""

from __future__ import annotations

import asyncio
import sys

from app.collections import web_collections as cols
from app.db import close_client, get_db


async def main() -> None:
    table_id = sys.argv[1] if len(sys.argv) > 1 else "T-0019"
    db = get_db()

    recs = await (
        db[cols.min_bet_recommendations]
        .find(
            {"tableId": table_id},
            projection={
                "_id": 0,
                "recommendationId": 1,
                "status": 1,
                "createdAt": 1,
                "drivers": 1,
                "currentMinBet": 1,
                "recommendedMinBet": 1,
            },
        )
        .sort("createdAt", -1)
        .to_list(length=5)
    )

    print(f"Recent recommendations for {table_id}:")
    if not recs:
        print("  (none)")
    for r in recs:
        occupancy_rate = (r.get("drivers") or {}).get("occupancyRate")
        created_at = r.get("createdAt")
        print(
            f"  {r.get('recommendationId')}  status={r.get('status')}  "
            f"current={r.get('currentMinBet')}→{r.get('recommendedMinBet')}  "
            f"occupancy={occupancy_rate}  createdAt={created_at}"
        )


async def _entrypoint() -> None:
    try:
        await main()
    finally:
        await close_client()


if __name__ == "__main__":
    asyncio.run(_entrypoint())
